src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -539,6 +591,57 @@ class IrsPlan extends React.Component<
     );
   }
 
+  /** getAllAssignments - funciton to get all assignments for the plan from OpenSRP
+   * @param {string} planId the id of the plan for which to return assignments
+   * @param {Organization[]} organizations an array of Organizations by which to query assignments
+   * @param {(assignments:Assignment[])=>any} callback a function handling the resulting Assignments
+   * @returns {(assignments:Assignment[])=>any} the callback passing the Assignments as the first parameter
+   */
+  public getAllAssignments(
+    planId: string,
+    organizations: Organization[],
+    callback: (assignments: Assignment[]) => any
+  ) {
+    // define a reference of all organization ids
+    const organizationIds: string[] = organizations.map((o: Organization) => o.identifier);
+
+    // generate api calls for each organization id
+    const assignmentPromises = organizationIds.map(
+      (o: string) =>
+        new Promise((resolve, reject) => {
+          OpenSrpAssignedService.read(o, { plan: planId })
+            .then(j => resolve(j))
+            .catch(error => reject(error));
+        })
+    );
+
+    // make all the calls to get assignments per org for the plan
+    return Promise.all(assignmentPromises).then((results: any[]) => {
+      const nextAssignments: Assignment[] = [];
+      // step through all api results per organization
+      for (let o = 0; o < results.length; o += 1) {
```

Comment:
I'm not sure of the best practices, but did you consider using `forEach` here?

```javascript
results.forEach(function(result, index) {
  ...
          organization: organizationIds[index],
  ...  
});
```

Reply:
@pld yup, I believe the good old `for` loop is typically quite a bit faster than the Array.prototype methods like `forEach` and `filter`, which is why I typically try to avoid them in files like this one where there can potentially be a LOT of processing of data.

Reply:
Oh an interesting spread of perf indicators https://github.com/dg92/Performance-Analysis-JS#results-for-mid-data-set-of-array-size-3000---20000